File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/game/ChangeGameTask.scala
package com.goexp.galgame.gui.task.game

import java.util

import com.goexp.galgame.gui.db.mongo.GameDB.{StarDB, StateDB}
import com.goexp.galgame.gui.model.Game
import javafx.concurrent.Task

object ChangeGameTask {

  class Like(private[this] val game: Game) extends Task[Void] {
    override protected def call: Void = {
      StateDB.update(game)
      null
    }
  }

  class MultiLike(private[this] val games: util.List[Game]) extends Task[Void] {

    override protected def call: Void = {
      StateDB.batchUpdate(games)
      null
    }
  }

  class MultiLikeByBrand(private[this] val brandId: Int) extends Task[Void] {

    override protected def call: Void = {
      StateDB.update(brandId)
      null
    }
  }

  class Star(private[this] val game: Game) extends Task[Void] {
    override protected def call: Void = {
      StarDB.update(game)
      null
    }
  }

}

File: desktopclient/src/main/scala/com/goexp/galgame/gui/task/game/ChangeGameTask.java
package com.goexp.galgame.gui.task.game;

import com.goexp.galgame.gui.db.mongo.GameDB;
import com.goexp.galgame.gui.model.Game;
import javafx.concurrent.Task;

import java.util.List;

public class ChangeGameTask {

    public static class Like extends Task<Void> {

        private Game game;

        public Like(Game game) {
            this.game = game;
        }

        @Override
        protected Void call() {
            new GameDB.StateDB().update(game);
            return null;
        }
    }

    public static class MultiLike extends Task<Void> {

        private GameDB.StateDB gameDB = new GameDB.StateDB();

        private List<Game> games;

        public MultiLike(List<Game> games) {
            this.games = games;
        }

        @Override
        protected Void call() {
            gameDB.batchUpdate(games);
            return null;
        }
    }

    public static class MultiLikeByBrand extends Task<Void> {

        private GameDB.StateDB gameDB = new GameDB.StateDB();

        private int brandId;

        public MultiLikeByBrand(int brandId) {
            this.brandId = brandId;
        }

        @Override
        protected Void call() {
            gameDB.update(brandId);
            return null;
        }
    }

    public static class Star extends Task<Void> {

        private Game game;

        public Star(Game game) {
            this.game = game;
        }

        @Override
        protected Void call() {
            new GameDB.StarDB().update(game);
            return null;
        }
    }
}
